tests/integration/route_test.py

```diff
@@ -0,0 +1,208 @@
+                    },
+                    'ipv6': {
+                        'address': [
+                            {
+                                'ip': IPV6_ADDRESS1,
+                                'prefix-length': 64
+                            }
+                        ],
+                        'dhcp': False,
+                        'autoconf': False,
+                        'enabled': True
+                    }
+                }
+            ],
+            'routes': {
+                'config': [
+                    IPV4_ROUTE1,
+                    IPV4_ROUTE2,
+                    IPV6_ROUTE1,
+                    IPV6_ROUTE2,
+                ]
+            },
+        })
+    cur_state = statelib.show_only(('eth1',))
+    _assert_has_route(IPV4_ROUTE1, cur_state)
+    _assert_has_route(IPV4_ROUTE2, cur_state)
+    _assert_has_route(IPV6_ROUTE1, cur_state)
+    _assert_has_route(IPV6_ROUTE2, cur_state)
+
+
+@pytest.mark.xfail(raises=NmstateNotImplementedError,
+                   reason="Red Hat Bug 1707396")
+def test_multiple_gateway(eth1_up):
+    netapplier.apply(
+        {
+            'interfaces': [
+                {
+                    'name': 'eth1',
+                    'state': 'up',
+                    'type': 'ethernet',
+                    'ipv4': {
+                        'address': [
+                            {
+                                'ip': IPV4_ADDRESS1,
+                                'prefix-length': 24
+                            }
+                        ],
+                        'dhcp': False,
+                        'enabled': True
+                    },
+                    'ipv6': {
+                        'address': [
+                            {
+                                'ip': IPV6_ADDRESS1,
+                                'prefix-length': 64
+                            }
+                        ],
+                        'dhcp': False,
+                        'autoconf': False,
+                        'enabled': True
+                    }
+                }
+            ],
+            'routes': {
+                'config': [
+                    IPV4_GATEWAY1,
+                    IPV4_GATEWAY2,
+                    IPV6_GATEWAY1,
+                    IPV6_GATEWAY2,
+                ]
+            },
+        })
+    cur_state = statelib.show_only(('eth1',))
+    _assert_has_route(IPV4_GATEWAY1, cur_state)
+    _assert_has_route(IPV4_GATEWAY2, cur_state)
+    _assert_has_route(IPV6_GATEWAY1, cur_state)
+    _assert_has_route(IPV6_GATEWAY2, cur_state)
+
+
+def _assert_has_route(route, state):
+    assert route in state['routes']['config']
+
+    # For running route, the default(-1) metric and route-table will be
+    # changed. Hence should be ignored.
+    tmp_route = copy.deepcopy(route)
+    run_routes = copy.deepcopy(state['routes']['running'])
+    for prop_name, default_value in (('metric', -1), ('table-id', 0)):
```

Comment:
I do not really understand where this scenario exists in the tests. The defaults seem to be checked against the desired state.
I do not expect the client to input in the desired state such values, as they mean nothing. If the client does not specify them, nmstate may internally use some value to represent it, but that is an implementation detail.
Is there something I am missing here?

Maybe you mean what is reported back in the 'config' state? I expect it will return either nothing or the actual explicit value. We cannot expect the client to have this logic.

Reply:
When desire state does not define metric or table_id. The config will be -1 and 0, the running will be unknown value. To do asserting, we need to assume above is legal.